server/db/driver/pg/internal/orders.go

```diff
@@ -14,6 +14,7 @@ const (
 		address TEXT,          -- INDEX
 		client_time TIMESTAMPTZ,
 		server_time TIMESTAMPTZ,
+		commit BYTEA, -- UNIQUE?
```

Comment:
One possible way to facilitate preventing reuse is by constraining booked orders to have a UNIQUE commit, which would make lookups in the orders tables quick, although doing so for the archived orders table would still be a burden given a long enough order history.

Reply:
Now the field is defined as `commit BYTEA UNIQUE`.

Reply:
Addressed in https://github.com/decred/dcrdex/pull/145/commits/66df38b6b690cb26ba7aa3d2dad9312596943b7a#diff-884dd3a776297ea8952dcb641648bd62R349-R353 where the new `db.ErrReusedCommit` is returned without killing the DB backend.